### DISCUSSION TITLE
Upsert Not Working as expected

### DISCUSSION THREAD

**Discussion by mitraraman** [supabase/supabase#19749] [Questions]
In one of our sync functions (executes every hour and can be executed manually as well) we add new customers and update customer data fields for customers already in our db. However, when testing locally I found that neither of these happened:
- one customer that didn't exist in the DB still doesn't exist
- one customer that existed but performed upsert with new info did not reflect new info in the DB

I didn't see any error messages either. Any suggestions?

**Comment by GaryAustin1** [comment]
Upsert can fail silently if you don't meet select RLS as it is an update call before it does an insert if the update conflict occurs.
For upsert to work in all cases you have to meet select,update and insert policies.

Also I would think both of your issues could be cause by your sync function not running for some reason or  doing the actual upsert.

Note it is unclear to me also if you mean you did lots of requests and 2 did not work out of those, or you are testing and those are two cases you tested that did not work.
